sycl/source/handler.cpp

```diff
@@ -285,6 +285,9 @@ event handler::finalize() {
             if (NewEvent != nullptr)
               NewEvent->setHostEnqueueTime();
             [&](auto... Args) {
+              if (MImpl->MKernelUsesClusterLaunch) {
+                Result = PI_ERROR_UNSUPPORTED_FEATURE;
```

Comment:
This doesn't seem enough since `Result` gets overwritten on L312. Also L291 should be `else if` to avoid the following PI calls in this case.

Reply:
Thanks for catching this, addressed in  (f0f9b)[https://github.com/intel/llvm/pull/14113/commits/f0f9bfdcfb2b98b0deb633aff547f5f014b1d0a5]